### ISSUE TITLE
Adding timedelta can produce malformed Times

### ISSUE THREAD

**Issue by MaxNoe** [astropy/astropy#4983]
Consider the following code:

``` python
from astropy.time import Time
import astropy.units as u
import numpy as np

times = Time('2015-01-01') + np.arange(0, 365) * u.day
times.to_datetime()
```

This fails, with the following error:

```
Traceback (most recent call last):
  File "example.py", line 6, in <module>
    times.to_datetime()
  File "/home/maxnoe/.local/anaconda3/lib/python3.5/site-packages/astropy/time/core.py", line 1416, in to_datetime
    return tm._shaped_like_input(tm._time.to_value(timezone))
  File "/home/maxnoe/.local/anaconda3/lib/python3.5/site-packages/astropy/time/formats.py", line 615, in to_value
    out[...] = datetime.datetime(iy, im, id, ihr, imin, isec, ifracsec)
ValueError: second must be in 0..59
```

This is because there is a malformed date in times:

```
'2015-06-30 23:59:60.000'
```

**Comment by mhvk** [comment]
Thanks for checking `Time`! In this case, though, you found a good reason for using `Time` and not `datetime`! `'2015-06-30 23:59:60.000'` is actually correct: it is a leap second. Sadly, `datetime`, like other unix times doesn't know what to do with leap seconds. Or, more accurately, I think it repeats `23:59:59` twice. Indeed, this fails too:
```
Time('2015-06-30 23:59:60.000').datetime
...
ValueError: second must be in 0..59
```

Which of course leaves the question of what to do...  I guess we could at least try to give a more informative error message. Beyond that, I think just converting to `59`, while correct in some sense, is asking for problems.  We could do what we do for `unix` time, which make all seconds a bit short on leap-second days.

@taldcroft - what do you think?

**Comment by mhvk** [comment]
Just adding a note: in python 3.6, datetime has acquired a `fold` attribute to help both leap seconds and daylight saving's time; see https://docs.python.org/3.6/whatsnew/3.6.html#whatsnew36-pep495

So, this could now be made to work for python 3.6, still raising a `ValueError` for earlier versions.

**Comment by ritwickdsouza** [comment]
I will attempt to solve this issue :)

**Comment by ritwickdsouza** [comment]
I have a open pull request, so how do i make a separate pull request with different commits since a repo can be forked only once?

**Comment by MaxNoe** [comment]
You can work in a second Branche in your Form and open the pull request for that branch

**Comment by bsipocz** [comment]
@ritwickdsouza - Please read the relevant sections in the [developer guide](http://docs.astropy.org/en/latest/development/workflow/development_workflow.html#astropy-guidelines-for-git), we strongly encourage everyone to always work on feature branches rather than on the master of their fork.

**Comment by MaxNoe** [comment]
Wow. You recommend renaming origin and upstream? That is not a good idea I think. It's opposite from any git and github tutorial I know and will probably confuse more than it helps.

**Comment by ritwickdsouza** [comment]
@MaxNoe Noted. Thanks :)
@bsipocz I'll go through the developer guide. Thanks. :)

**Comment by bsipocz** [comment]
Well, I find this way super useful and logical. One work with multiple remotes and open PRs against other people's forks (in that scenario those are the "upstream") as well as PRs against the main repo, and also sometimes push changes directly to master. This way origin is always my own github name and doesn't cause confusion, upstream then the name of the main repo.  That said I didn't read many git tutorials before reading this astropy one a few years ago and it served me well from being a novice to being a maintainer of packages.

Anyway the main point for now is not to work on user(origin)/master ever,  but use feature branches.

**Comment by mhvk** [comment]
I'm closing this issue, since we now have an appropriate error message. I looked into whether we now can use the `fold` attribute, and in principle it is possible, but `datetime` still very much does not support leap seconds (e.g., it ignores the fold attribute in comparisons), so I think it is better to just raise.
